src/main/scala/firrtl/passes/ExpandWhens.scala

```diff
@@ -0,0 +1,189 @@
+
+  case class RefNotInitializedException(info: Info, mname: String, name: String) extends PassException(
+      s"$info : [module $mname]  Reference $name is not fully initialized.")
+
+  // ========== Expand When Utilz ==========
+  private def getEntries(
+      hash: LinkedHashMap[WrappedExpression, Expression],
+      exps: Seq[Expression]): LinkedHashMap[WrappedExpression, Expression] = {
+    val hashx = LinkedHashMap[WrappedExpression, Expression]()
+    exps foreach (e => if (hash.contains(e)) hashx(e) = hash(e))
+    hashx
+  }
+  private def getFemaleRefs(n: String, t: Type, g: Gender): Seq[Expression] = {
+    def getGender(t: Type, i: Int, g: Gender): Gender = times(g, get_flip(t, i, DEFAULT))
+    val exps = create_exps(WRef(n, t, ExpKind(), g))
+    val expsx = ArrayBuffer[Expression]()
+    for (i <- 0 until exps.size) {
+      getGender(t, i, g) match {
+        case (BIGENDER | FEMALE) => expsx += exps(i)
+        case _ =>
+      }
+    }
+    expsx
+  }
+  private def squashEmpty(s: Stmt): Stmt = {
+    val res = s match {
+      case Begin(stmts) => Begin(stmts filter (_ != Empty()))
+      case s => s
+    }
+    res map squashEmpty
+  }
+
+
+  // ------------ Pass -------------------
+  def run(c: Circuit): Circuit = {
+    def expandWhens(m: InModule): (LinkedHashMap[WrappedExpression, Expression], ArrayBuffer[Stmt], Stmt) = {
+      val namespace = Namespace(m)
+      val simlist = ArrayBuffer[Stmt]()
+      def expandWhens(netlist: LinkedHashMap[WrappedExpression, Expression], p: Expression)(s: Stmt): Stmt = {
+        s match {
+          case w: DefWire =>
+            val ref = WRef(w.name, w.tpe, WireKind(), BIGENDER)
+            getFemaleRefs(w.name, w.tpe, BIGENDER) foreach (ref => netlist(ref) = WVoid())
+            w
+          case r: DefRegister =>
+            getFemaleRefs(r.name, r.tpe, BIGENDER) foreach (ref => netlist(ref) = ref)
+            r
+          case c: Connect =>
+            netlist(c.loc) = c.exp
+            Empty()
+          case c: IsInvalid =>
+            netlist(c.exp) = WInvalid()
+            Empty()
+          case s: Conditionally =>
+            val exps = ArrayBuffer[Expression]()
+            val memos = ArrayBuffer[Stmt]()
+
+            def prefetch(s: Stmt): Stmt = s match {
+              case s: Connect => exps += s.loc; s
+              case s => s map prefetch
+            }
+            prefetch(s.conseq)
+
+            val c_netlist = getEntries(netlist, exps)
+            val conseqStmt = expandWhens(c_netlist, AND(p, s.pred))(s.conseq)
+            val altStmt = expandWhens(netlist, AND(p, NOT(s.pred)))(s.alt)
+
+            for (lvalue <- c_netlist.keys) {
+              val value = netlist.get(lvalue) match {
+                case v: Some[Expression] => v.get match {
+                  case WVoid() =>
+                    val decl = getDeclaration(m, lvalue.e1)
+                    throw new RefNotInitializedException(decl.info, m.name, decl.name)
```

Comment:
This check isn't correct - I'll commit a test case that shows it. You need to check both the consequence and the alternative branches for the WVoid(), instead of just the consequent.